File: ruby/lib/Sikaru/types/agent_import_eval_suite_request.rb
# frozen_string_literal: true

module Sikaru
  module Types
    class AgentImportEvalSuiteRequest < Internal::Types::Model
      field :kind, -> { String }, optional: false, nullable: false

      field :metadata, -> { Internal::Types::Hash[String, Object] }, optional: true, nullable: false

      field :primary_metric, -> { String }, optional: false, nullable: false, api_name: "primaryMetric"

      field :storage_ref, -> { String }, optional: false, nullable: false, api_name: "storageRef"
    end
  end
end

File: python/src/sikaru_api/types/agent_import_eval_suite_request.py

import typing

import pydantic
import typing_extensions
from ..core.pydantic_utilities import IS_PYDANTIC_V2, UniversalBaseModel
from ..core.serialization import FieldMetadata


class AgentImportEvalSuiteRequest(UniversalBaseModel):
    kind: str
    metadata: typing.Optional[typing.Dict[str, typing.Any]] = None
    primary_metric: typing_extensions.Annotated[
        str, FieldMetadata(alias="primaryMetric"), pydantic.Field(alias="primaryMetric")
    ]
    storage_ref: typing_extensions.Annotated[str, FieldMetadata(alias="storageRef"), pydantic.Field(alias="storageRef")]

    if IS_PYDANTIC_V2:
        model_config: typing.ClassVar[pydantic.ConfigDict] = pydantic.ConfigDict(extra="allow", frozen=True)  # type: ignore # Pydantic v2
    else:

        class Config:
            frozen = True
            smart_union = True
            extra = pydantic.Extra.allow
